src/veins/modules/mobility/traci/TraCIScenarioManager.cc

```diff
@@ -249,6 +250,15 @@ void TraCIScenarioManager::initialize(int stage)
     ignoreGuiCommands = par("ignoreGuiCommands");
     host = par("host").stdstringValue();
     port = par("port");
+    if (port == -1) {
+        // cascaded search for externally configured traci port
+        const char* env_port = std::getenv("VEINS_TRACI_PORT");
+        if (env_port == nullptr) {
+            error("TraCI Port autoconfiguration failed, set `port` != -1 in omnetpp.ini or provide VEINS_TRACI_PORT environment variable.");
+        }
+        port = std::atoi(env_port);
+    }
+    ASSERT(port > 0 && port < 65536);
```

Comment:
This ASSERT kills `TraCIScenarioManagerForker`, which supports setting port == -1 for autodetection. See <https://github.com/sommer/veins/blob/veins-5a1/src/veins/modules/mobility/traci/TraCIScenarioManagerForker.ned#L41>

Reply:
Then we'll leave it out.
I was wondering how we could identify problems with converting `VEINS_TRACI_PORT` into an int (`std::atoi` returns 0 on failure). But maybe keeping it flexible here is the better way.

Reply:
Sorry, my bad. Even with the `ASSERT` removed, `TraCIScenarioManagerForker` still fails when autodetection is turned on. I'm afraid you'll have to test yourself to see exactly why.